### ISSUE TITLE
[Accessibility]: Multiple spans create reading challenges for screen readers

### ISSUE THREAD

**Issue by GarlonJG** [department-of-veterans-affairs/vets-design-system-documentation#5101]
### Point of contact

Garlon Garth @GarlonJG, or Heather Rienks

### Defect Severity

1, Launchblocker. Critical. Must be fixed before launch.

### Severity rationale

Chose this level due to a finding from a 508 Audit Report, and it causes reading issues that make it hard to understand what exactly is being read.

### VA 508 Office

Yes this is from a 508 Audit Report. Original issue found here [Defect 2](https://github.com/department-of-veterans-affairs/va.gov-team/issues/120036).

### URL where issue occurs

https://staging.va.gov

### Component or pattern (if applicable)

va-segmented-progress-bar

### Environment details

- OS: Mac 26.0.1
- Browser: Chrome
- Screen Reader: MacVO (JAWS also noted from this Slack convo [Slack convo](https://dsva.slack.com/archives/C8E985R32/p1761669399549529))
- Device: MacBook Pro M1

### Steps to reproduce

1. Log into staging user 81
2. Navigate to https://staging.va.gov/manage-va-debt/summary/debt-balances
3. Click "Resolve this debt" for any debt/overpayment
4. Click "Dispute an overpayment online"
5. Click "Continue your application"
6. Use any screen reader to read the header in the va-segmented-progress-bar

### Observed behavior

When multiple spans are used in focusable elements for screen readers, they can cause reading issues. In JAWS, multiple spans collapse spaces between spans causing run-on or new combined words. In MacVO, it causes the header level to be re-read for each span in the focusable element.

### Expected behavior / Acceptance criteria

The expected behavior is that screen readers should produce a clear, continuous, and coherent reading of headers or focusable elements, avoiding breaks or unnecessary repetition.

### Supporting evidence (screenshots, recordings, code snippets)

<img width="695" height="206" alt="Image" src="https://github.com/user-attachments/assets/8061ba76-1774-4713-8575-1f2bda9b1676" />

### Related issues or systemic notes

It goes without saying that I'm sure this issue is prevalent in other components where design elements are used by screen readers to convey messages to the user. I have a branch pushed up with the fix for reference that seems to solve the issue [120036-GG](https://github.com/department-of-veterans-affairs/component-library/tree/120036-GG). The fix separates sr-only text from visual design elements.

### Type of issue

- [x] Axe-core
- [ ] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [x] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [x] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [x] Overall user experience
- [x] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else

**Comment by jeana-adhoc** [comment]
I've done some digging into the work and problem here..

**Some notes:**
1. This issue happens because WebKit browsers see `<span>` as individual items in the dom. So the progress stepper is seen as having 4 separate items.
<img width="478" height="129" alt="Image" src="https://github.com/user-attachments/assets/db87e25b-6898-43e5-8576-a77041391cb3" />
2. The issue is componded because each `<span>` also has its own css class on it. (Sometimes VO won't merge them if they have the same computed styles across all the spans but that depends on safari and macos versions).
3. There are some suggestions that we could put `role="text"` on the parent but that is only a safari/voice over supported solution, and `role="text"` is not supported in the current ARIA spec.

Here is a before/after of what VO sounds like before and after

**Before:**


**After:**

I think this is a solid approach from Accessibility.  Next step to have a Design System Engineer review the code.  You can [see the diff in this commit](https://github.com/department-of-veterans-affairs/component-library/commit/eba03abb97604af0d0b6dd5bf04b4c0faf96d8b9)

**Comment by jeana-adhoc** [comment]
I reviewed the PR, and marked it good to merge!

Tested this with:

✅ Mac:

Chrome + VO: "Heading level 4, You are on step 2 of 5 {Step Name}"
Safari + VO: "Heading level 4, You are on step 2 of 5 {Step Name}"
Firefox + VO: "Heading level 4, You are on step 2 of 5 {Step Name}"

✅ Windows:

Edge + NVDA: "You are on step two of 5 {step name}, heading level 4"
Edge + JAWS: "You are on step two of 5 {step name}, heading level 4"
Firefox + NVDA: "You are on step two of 5 {step name}, heading level 4"
Firefox + JAWS: "You are on step two of 5 colon {step name}, heading level 4"

There are no regressions, and VO sounds so much better.
